Add support to the CLI command `run` for converting a config from a different format (#4295)
grafana/agent: docs/sources/flow/reference/cli/run.md

to advertise from a list of default network interfaces. The agent must be
reachable over HTTP on this address as communication happens over the agent's
HTTP server.

> ## Converter mode

Reviewer: "Mode" tends to be reserved for operational modes (static mode, flow mode), so I don't want to overload the term here. 

Can we just call it what it is? Something like "Configuration conversion?"

Author: done

to advertise from a list of default network interfaces. The agent must be
reachable over HTTP on this address as communication happens over the agent's
HTTP server.

## Configuration conversion (beta)

When the `--config.format` command-line argument is provided, Grafana
Agent will convert the config file from the source format to River and
immediately start running with it. This leverages the same converter API
described in the [grafana-agent convert][] docs.

> If the `--config.bypass-conversion-warnings` command-line argument is also provided,
> Grafana Agent will ignore any warnings provided by the converter. This should
> be used with caution since the resulting conversion is not equivalent to the
> original config.

Reviewer 2: ```suggestion
If you also use the `--config.bypass-conversion-warnings` command-line argument,
Grafana Agent will ignore any warnings from the converter. Use this
argument with caution because the resulting conversion is not equivalent to the
original configuration.
```
